### PR TITLE
Replaces early pam_pwhistory logic with method suitable for 8.8+

### LINKED ISSUE #435: [BUG] RHEL 8.8+ and 9.2+ Require Use of `authselect` to enable `pam_pwhistory.so`
**Describe the bug**

Current enablement of `pam_pwhistory.so` in the `/etc/pam.d/password-auth` and `/etc/pam.d/system-auth` files is effected through direct-edits of the files via [cat2/RHEL-08-020220.sls](https://github.com/plus3it/ash-linux-formula/blob/master/ash-linux/el8/STIGbyID/cat2/RHEL-08-020220.sls) and [cat2/RHEL-08-020221.sls](https://github.com/plus3it/ash-linux-formula/blob/master/ash-linux/el8/STIGbyID/cat2/RHEL-08-020221.sls), respectively. Per updated [vendor-guidance](https://access.redhat.com/solutions/6980935), the `/etc/pam.d` entries should be managed through the `authselect` utility and associated module-configuration files.

**Severity**

- [ ] Completely Broken (No work-around evident)
- [ ] Severely Broken (Work-around possible but difficult)
- [X] Moderately Broken (Trivial work-around)
- [ ] Nuisance (Functions but untrapped errors can slip through)

**To Reproduce**
1. Launch an EL8 EC2
2. Run watchmaker, inclusive of using the join-domain formula
3. Audit the system for `pam_pwhistory.so` compliance:
    - `pam_pwhistory.so` will be missing from `/etc/pam.d/password-auth`
    - `pam_pwhistory.so` will be missing from `/etc/pam.d/system-auth`
    - An error message similar to the following will appear in `journald`
        ~~~
        Aug 07 15:12:44 clobber-test realmd[17975]:  * /usr/bin/sh -c /usr/bin/authselect select sssd with-mkhomedir --force && /usr/bin/systemctl enable oddjobd.service && /usr/bin/systemctl start oddjobd.service
        Aug 07 15:12:44 clobber-test realmd[17975]: process started: 18034
        Aug 07 15:12:44 clobber-test realmd[17975]: [error] [/etc/authselect/system-auth] has unexpected content!
        Aug 07 15:12:44 clobber-test realmd[17975]: [error] [/etc/authselect/password-auth] has unexpected content!
        Aug 07 15:12:44 clobber-test realmd[17975]: [error] [/etc/authselect/postlogin] has unexpected content!
        Aug 07 15:12:44 clobber-test realmd[17975]: Backup stored at /var/lib/authselect/backups/2023-08-07-15-12-44.Rlbd5S
        Aug 07 15:12:44 clobber-test realmd[17975]: Profile "sssd" was selected.
        Aug 07 15:12:44 clobber-test realmd[17975]: The following nsswitch maps are overwritten by the profile:
        Aug 07 15:12:44 clobber-test realmd[17975]: - passwd
        Aug 07 15:12:44 clobber-test realmd[17975]: - group
        Aug 07 15:12:44 clobber-test realmd[17975]: - netgroup
        Aug 07 15:12:44 clobber-test realmd[17975]: - automount
        Aug 07 15:12:44 clobber-test realmd[17975]: - services
        ~~~

**Expected behavior**

Upon completion of all watchmaker states, the `pam_pwhistory.so` module is properly present in the `/etc/pam.d/password-auth` and `pam_pwhistory.so` files.

**Deviance Description**

**Screenshots**

**Additional context**

**Fix Suggestions**

Update the `…/cat2/RHEL-08-020220.sls` and `…/cat2/RHEL-08-020221.sls` files to adhere to current vendor-guidance.

### PR DESCRIPTION
**Pull request by ferricoxide**
Closes #435

### CODE REVIEW

File: ash-linux/el8/STIGbyID/cat2/RHEL-08-020220.sls [hunk 1]
```diff
@@ -39,27 +37,50 @@ notify_{{ stig_id }}-skipSet:
...
-    - pattern: '^(?P<srctok>{{ searchRoot }}.*$)'
-    - repl: '\g<srctok> remember=5'
-  # Entry is missing
-  {%- else %}
-file_{{ stig_id }}-{{ targFile }}:
+    - name: '{{ pwhistory_cfg_file }}'
+    - append_if_not_found: True
+    - not_found_content: |-
+
+        # Inserted per STIG IDs RHEL-08-020220 and RHEL-08-020221
```
[the remark this answers is not in the document]
Comment (ferricoxide) [reply ash-linux/el8/STIGbyID/cat2/RHEL-08-020220.sls:66]: Using the `authselect` method subsumes _both_ RHEL-08-020220 _and_  RHEL-08-020221.  When using `authselect` I don't know that you can say "_only_ update `/etc/pam.d/system-auth`" or "_only_ update `/etc/pam.d/password-auth`. So, I had to collapse them into a single file with a dual-reference (at least until they fix the STIGs to reflect the change in management of the `/etc/pam.d` files).

File: ash-linux/el7/STIGbyID/cat2/files/RHEL-07-020023.sh [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#!/bin/bash
+#
```
[the remark this answers is not in the document]
Comment (ferricoxide) [reply ash-linux/el7/STIGbyID/cat2/files/RHEL-07-020023.sh:2]: Is hers going to be merged first? This is a quick PR that I need to get in so I can address the `pam_faillock`  module.

File: ash-linux/el8/STIGbyID/cat2/RHEL-08-020220.sls (hunk 1 above)
[the remark this answers is not in the document]
Comment (eemperor) [reply ash-linux/el8/STIGbyID/cat2/RHEL-08-020220.sls:66]: Okay.  Good for me.

File: ash-linux/el7/STIGbyID/cat2/files/RHEL-07-020023.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (eemperor) [reply ash-linux/el7/STIGbyID/cat2/files/RHEL-07-020023.sh:2]: That's fine

**Review by eemperor — APPROVED** [review]

**Pull request merged**